src/st/nz0/3E30C.c

```diff
@@ -559,8 +559,28 @@ void func_801C07FC(Entity* entity) {
     }
 }
 
-// DECOMP_ME_WIP func_801C090C https://decomp.me/scratch/0VI4v
-INCLUDE_ASM("asm/us/st/nz0/nonmatchings/3E30C", func_801C090C);
+void func_801C090C(
+    Entity* self, u8 count, u8 params, s32 x, s32 y, u8 arg5, s16 xGap) {
+    Entity* newEntity;
+    s32 i;
+    s16 newX = self->posX.i.hi + x;
+    s16 newY = self->posY.i.hi + y;
+
+    for (i = 0; i < count; i++) {
+        newEntity = AllocEntity(&g_Entities[160], &g_Entities[192]);
+        if (newEntity != NULL) {
+            newEntity->posX.i.hi = newX + xGap * i;
+            newEntity->posY.i.hi = newY;
+            newEntity->objectId = 0x14;
```

Comment:
Use the enum value? https://github.com/Xeeynamo/sotn-decomp/blob/6398c1f18ed890f2a5a03a50d5c0c274a0bf5d48/src/st/nz0/nz0.h#L3